src/main/kotlin/io/snyk/plugin/ui/jcef/IgnoreInFileHandler.kt

```diff
@@ -0,0 +1,96 @@
+package io.snyk.plugin.ui.jcef
+
+import com.intellij.openapi.diagnostic.LogLevel
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
+import com.intellij.ui.jcef.JBCefBrowserBase
+import com.intellij.ui.jcef.JBCefJSQuery
+import io.snyk.plugin.runInBackground
+import io.snyk.plugin.ui.SnykBalloonNotificationHelper
+import org.cef.browser.CefBrowser
+import org.cef.browser.CefFrame
+import org.cef.handler.CefLoadHandlerAdapter
+import snyk.common.IgnoreService
+import snyk.common.lsp.LanguageServerWrapper
+import java.io.File
+import java.io.IOException
+
+class IgnoreInFileHandler(
+    private val project: Project,
+    ) {
+    val logger = Logger.getInstance(this::class.java).apply {
+        // tie log level to language server log level
+        val languageServerWrapper = LanguageServerWrapper.getInstance()
+        if (languageServerWrapper.logger.isDebugEnabled) this.setLevel(LogLevel.DEBUG)
+        if (languageServerWrapper.logger.isTraceEnabled) this.setLevel(LogLevel.TRACE)
+    }
+
+    fun generateIgnoreInFileCommand(jbCefBrowser: JBCefBrowserBase): CefLoadHandlerAdapter {
+        val applyIgnoreInFileQuery = JBCefJSQuery.create(jbCefBrowser)
+
+        applyIgnoreInFileQuery.addHandler { value ->
+            val params = value.split("|@", limit = 2)
+            val issueId = params[0] // ID of issue that needs to be ignored
+            val filePath = params[1]
+            // Computed path that will be used in the snyk ignore command for the --path arg
+            val computedPath = filePath.removePrefix("${project.basePath}${File.separator}")
```

Comment:
is it necessary to make the file relative?

Reply:
`project.basePath` should not be used in new code - instead the content root path should be used.

Reply:
Yes, it doesn't work with the full path, it seems that the `snyk ignore` expects the path to be relative to the project root folder.

Reply:
I'll use the `getContentRootPaths()` instead of the `basePath`